### PR TITLE
condition added for cross icon.

### LINKED ISSUE #368: A conditional to be added in Angular Material search dialog for the `x` in the search field
# Angular App version

- [ ] `angular-v17` - has no UI libraries
- [x] `angular-v17-AnguMAT` - has Angular Material
- [ ] `angular-v17-PrimeNG` - has PrimeNG 

**Additional context**
The clear input `x` in the search should only appear if there is some value entered in the search field 
so just a conditional if to check for value in the template needs to be added 

image

### PR DESCRIPTION
**Pull request by Ali7040**
# Angular App version

- [ ] `angular-v17` - has no UI libraries
- [x] `angular-v17-AnguMAT` - has Angular Material
- [ ] `angular-v17-PrimeNG` - has PrimeNG

## This PR Closes Issue 
closes #368

## What type of PR is this? (check all applicable)
- [ ] 🅰️ Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Mobile & Desktop Screenshots/Recordings
[Attach screenshots or recordings if applicable]

## Added to documentation?
- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

## [Optional] Post-deployment tasks
[Specify any post-deployment tasks that need to be performed]

## [Optional] What gif best describes this PR or how it makes you feel? 
[Embed gif or describe the feeling in plain text]

### CODE REVIEW

File: angular-v17-AnguMAT/src/app/partials/search-dialog/search-dialog.component.html [hunk 1]
```diff
@@ -5,35 +5,34 @@ <h2>Search posts</h2>
...
-        <mat-icon class="">search</mat-icon>
-        <p>Search articles from this blog</p>
-      </div>
-    }
-  }
+  <div class="no-result-text">
+    <mat-icon class="">search</mat-icon>
+    <p>Search articles from this blog</p>
+  </div>
+  } }
```
Comment (cleancodecraft) [inline angular-v17-AnguMAT/src/app/partials/search-dialog/search-dialog.component.html:37]: @Ali7040 I said this to another contributor earlier - please pay attention to formatting & indentation when there is so much change that is only indentation it is hard to see what are the real changes that you made.

You shared on discord that you contributed to another open source project did you give the PRs with changed formatting like this too ?

File: angular-v17-AnguMAT/src/app/partials/search-dialog/search-dialog.component.html [hunk 2]
```diff
@@ -5,35 +5,34 @@ <h2>Search posts</h2>
...
-          <p class="post-info">{{ post.publishedAt | date }}</p>
-        </div>
+  @if (queryFormControl.value && posts.length !== 0) { @for (post of posts;
+  track post.id) {
+  <div class="search-result" (click)="navigateToPost(post.slug)">
+    <div class="info-section">
+      <h2 class="post-title">{{ post.title }}</h2>
+      <p class="author-name post-info">{{ post.author.name }}</p>
+      <p class="post-info">{{ post.publishedAt | date }}</p>
+    </div>
```
Comment (cleancodecraft) [inline angular-v17-AnguMAT/src/app/partials/search-dialog/search-dialog.component.html:23]: @Ali7040 I said this to another contributor earlier - please pay attention to formatting & indentation when there is so much change that is only indentation it is hard to see what are the real changes that you made.

You shared on discord that you contributed to another open source project did you give the PRs with changed formatting like this too ?

Comment (cleancodecraft) [reply angular-v17-AnguMAT/src/app/partials/search-dialog/search-dialog.component.html:23]: @Ali7040 please fix the formatting to be like the original, only include your changed code and then we can merge, the work you did its ok

**Review by cleancodecraft — CHANGES_REQUESTED** [review]
@Ali7040 Fix formatting and indentation as explained in comments

**Comment by Ali7040** [comment]
> @Ali7040 Fix formatting and indentation as explained in comments
@cleancodecraft Firstly, I am using the VS Code extension Prettier - Code Formatter, and I don't think it's a problem.

**Comment by Ali7040** [comment]
> @Ali7040 Fix formatting and indentation as explained in comments

any other problem.

File: angular-v17-AnguMAT/src/app/partials/search-dialog/search-dialog.component.html [hunk 3]
```diff
@@ -5,35 +5,34 @@ <h2>Search posts</h2>
 
 <mat-dialog-content>
   <mat-form-field class="search-input">
-    <input matInput type="text" [formControl]="queryFormControl">
+    <input matInput type="text" [formControl]="queryFormControl" />
     <button matSuffix mat-icon-button aria-label="Clear" (click)="clearQuery()">
+      @if(queryFormControl.value){
```
[the remark this answers is not in the document]
Comment (Ali7040) [reply angular-v17-AnguMAT/src/app/partials/search-dialog/search-dialog.component.html:10]: @cleancodecraft, I don't need to tell others how many I did, and I use Prettier - Code Formatter; it automatically does formatting. If you want to review the code, do it. You don't need to leave comments. I mentioned in the PR title that I added a condition.

**Comment by cleancodecraft** [comment]
@Ali7040 you have been removed from the collaborators list since I appealed to Esther to remove you for not following the guidelines - multiple times now !

# READ THE [CODING RULES](https://github.com/monacodelisa/Angular-Headless-Hashnode/blob/develop/CONTRIBUTING.md#coding-rules)

- understand that we are not here to accommodate you and what you want to do or what you feel is right 
- you are making changes to the code which are not acceptable
- if you cannot follow super simple guidelines, this project is not for you
- have a great day

**Pull request closed without merging**